test/factories/delegatedManagerFactory.spec.ts

```diff
@@ -0,0 +1,707 @@
+
+      const delegatedManager = await deployer.manager.getDelegatedManager(initializeParams.manager);
+
+      expect(await delegatedManager.setToken()).eq(setTokenAddress);
+      expect(await delegatedManager.factory()).eq(delegatedManagerFactory.address);
+      expect(await delegatedManager.methodologist()).eq(subjectMethodologist);
+      expect(await delegatedManager.useAssetAllowlist()).eq(true);
+    });
+
+    it("should set the intialization state correctly", async() => {
+      const createdContracts = await delegatedManagerFactory.callStatic.createSetAndManager(
+        subjectComponents,
+        subjectUnits,
+        subjectName,
+        subjectSymbol,
+        subjectOwner,
+        subjectMethodologist,
+        subjectModules,
+        subjectOperators,
+        subjectAssets,
+        subjectExtensions
+      );
+
+      const tx = await subject();
+
+      const setTokenAddress = await protocolUtils.getCreatedSetTokenAddress(tx.hash);
+      const initializeParams = await delegatedManagerFactory.initializeState(setTokenAddress);
+
+      expect(initializeParams.deployer).eq(owner.address);
+      expect(initializeParams.owner).eq(subjectOwner);
+      expect(initializeParams.isPending).eq(true);
+      expect(initializeParams.manager).eq(createdContracts[1]);
+    });
+
+    it("should emit a DelegatedManagerDeployed event", async() => {
+      const createdContracts = await delegatedManagerFactory.callStatic.createSetAndManager(
+        subjectComponents,
+        subjectUnits,
+        subjectName,
+        subjectSymbol,
+        subjectOwner,
+        subjectMethodologist,
+        subjectModules,
+        subjectOperators,
+        subjectAssets,
+        subjectExtensions
+      );
+
+      await expect(subject()).to.emit(delegatedManagerFactory, "DelegatedManagerCreated").withArgs(
+        createdContracts[0], // SetToken
+        createdContracts[1], // DelegatedManager
+        owner.address
+      );
+    });
+
+    describe("when the assets array is non-empty but missing some component elements", async() => {
+      beforeEach(async() => {
+        subjectAssets = [setV2Setup.dai.address];
+      });
+
+      it("should revert", async() => {
+        await expect(subject()).to.be.revertedWith("Asset list must include all components");
+      });
+    });
+
+    // This test exists to cover the condition on the branch which validates components are assets
+    // Just verifying that creation state is pending
+    describe("when the assets array is empty", async() => {
+      beforeEach(() => {
+        subjectAssets = [];
+      });
+
+      it("should set the intialization state correctly", async() => {
+        const tx = await subject();
+
+        const setTokenAddress = await protocolUtils.getCreatedSetTokenAddress(tx.hash);
+        const initializeParams = await delegatedManagerFactory.initializeState(setTokenAddress);
+
+        expect(initializeParams.isPending).eq(true);
+      });
```

Comment:
Add a test to verify that DelegateManager's `useAssetAllowlist` is false

Reply:
[1901b6a](https://github.com/SetProtocol/set-v2-strategies/pull/14/commits/1901b6ac4ebe9007336d6b8621ba35ea424477fb)